app.js

```diff
@@ -0,0 +1,173 @@
+    }, err => {
+        if (err) {
+            debug('Error starting MongoStore: %s'.red, err);
+        }
+    });
+
+    mongoStore.on('error', err => {
+        debug('Error with MongoStore: %s'.red, err);
+    });
+
+    // Express configuration
+    app.set('view engine', 'ejs');
+    app.use(express.static(__dirname + '/css'));
+    app.use(cookieParser());
+    app.use(bodyParser.json({ extended: false }));
+    app.use(bodyParser.urlencoded({ extended: false }));
+    app.use(errorHandler());
+    app.use(session({ secret: config.sessionsecret, resave: false, saveUninitialized: false, store: mongoStore }));
+    app.use(passport.initialize());
+    app.use(passport.session());
+
+    // Routes
+    app.get('/', routes.site.index);
+    app.get('/login', routes.site.loginForm);
+    app.post('/login', routes.site.login);
+
+    app.get(config.oauth.authorizationPath, routes.oauth2.authorization);
+    app.post(config.oauth.tokenPath, routes.oauth2.token);
+
+    //Create test users in database
+    debug('Creating test users: %O', config.testUsers);
+
+    const db = mongojs('localhost/' + config.mongo.database, ['users']);
+
+    function saveUserAsync(user) {
+        return new Promise(function(resolve, reject) {
+            db.users.save(user, function (err, doc) {
+                if (err) {
+                    return reject(err);
+                }
+                return resolve([doc]);
+            });
+        });        }
+
+    let promises = [
+        saveUserAsync(config.testUsers.authorUser),
+        saveUserAsync(config.testUsers.editorUser),
+        saveUserAsync(config.testUsers.adminUser)
+    ];
+
+    Promise.all(promises)
+        .then(function(allData) {
+            debug('Successfully added test users to the database %s', config.mongo.database);
+            try {
+                app.listen(config.net.port, () => {
+                    debug('guestlister %s with API version %s waiting for requests on port %s',
+                        config.version,
+                        config.api_version,
+                        config.net.port);
+                });
+
+            } catch (err) {
+                callback(err);
+            }
+        });
+
+    callback(null);
+}
+
+// auto_reconnect is on by default and only for RE(!)connects, BUT not for the initial attempt: http://bites.goodeggs.com/posts/reconnecting-to-mongodb-when-mongoose-connect-fails-at-startup/
+const dbBackoff = backoff.fibonacci({
+    randomisationFactor: 0,
+    initialDelay: config.mongo.initial_connection_initial_delay,
+    maxDelay: config.mongo.initial_connection_max_delay
+});
+
+dbBackoff.failAfter(config.mongo.initial_connection_attempts);
+dbBackoff.on('backoff', function (number, delay) {
+    debug('Trying to connect to MongoDB in %sms', delay);
+});
+dbBackoff.on('ready', function (number, delay) {
+    debug('Connect to MongoDB (#%s) ...', number);
+    mongoose.connect(dbURI, {
+        promiseLibrary: global.Promise
```

Comment:
Configure mongoose only with one object, not here and in https://github.com/o2r-project/o2r-guestlister/pull/1/files#diff-0364f57fbff2fabbe941ed20c328ef1aR39

See `dbOptions` object at https://github.com/o2r-project/o2r-muncher/blob/master/index.js#L39

Reply:
done